apps/sensenet/cypress/integration/ui/addNewMenu.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { pathWithQueryParams } from '../../../src/services/query-string-builder'
+
+context('AddNew Menu', () => {
+  before(() => cy.clearCookies({ domain: null } as any))
+
+  beforeEach(() => {
+    cy.login()
+  })
+
+  it('should open a dropdown with the list of allowed child types', () => {
+    const dropdownItems = [
+      'Folder',
+      'Document Library',
+      'Image Library',
+      'Event List',
+      'Memo List',
+      'Link List',
+      'Task List',
+      'Custom List',
+      'Workspace',
+      'System Folder',
+      'Demo Workspace',
+    ]
+    cy.visit(pathWithQueryParams({ path: '/', newParams: { repoUrl: Cypress.env('repoUrl') } }))
+      .get('a[href="/content/explorer/"]')
+      .click()
+      .get('span[title="Add new"]')
+      .click()
+      .get('ul.MuiList-root > div.MuiListItem-root > span.MuiListItemText-primary')
+      .each(($span) => {
+        const text = $span.text()
+        if (text) {
+          expect(dropdownItems).to.include(text)
+        }
+      })
+  })
+
+  it('should display an editor of new content and AddNew button should be disabled after selection', () => {
+    cy.visit(pathWithQueryParams({ path: '/', newParams: { repoUrl: Cypress.env('repoUrl') } }))
+      .get('a[href="/content/explorer/"]')
+      .click()
+      .get('span[title="Add new"]')
+      .click()
+      .get('ul.MuiList-root > div.MuiListItem-root > span.MuiListItemText-primary')
```

Comment:
This could be also selected by adding an additional data attribute (data-test="listitem") to the element in the component/AddButtons.tsx file (he list starts somewhere at line 217.)